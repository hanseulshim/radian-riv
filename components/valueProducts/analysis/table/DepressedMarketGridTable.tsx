import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import { formatPercent } from 'utils'

interface Props {
  tableData: any[]
}

export default function DepressedMarketGridTable({ tableData }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])
  const fetchData = useCallback(async () => {
    setColumns([
      {
        Header: '',
        accessor: 'category',
        width: 200,
        align: 'right'
      },
      {
        Header: '# of Listings',
        accessor: 'listingsCount',
        align: 'right',
        width: 125
      },
      {
        Header: '% of Listings',
        accessor: row => {
          return formatPercent(row.listingsPercent)
        },
        align: 'right'
      },
      {
        Header: '# of Pendings',
        accessor: 'pendingsCount',
        align: 'right',
        width: 125
      },
      {
        Header: '% of Pendings',
        accessor: row => {
          return formatPercent(row.pendingsPercent)
        },
        align: 'right'
      },
      {
        Header: '# of Solds',
        accessor: 'soldsCount',
        align: 'right',
        width: 125
      },
      {
        Header: '% of Solds',
        accessor: row => {
          return formatPercent(row.soldsPercent)
        },
        align: 'right'
      }
    ])
    setData(tableData)
  }, [tableData])

  return (
    <div style={{ marginTop: '1em' }}>
      <Table columns={columns} data={data} fetchData={fetchData} />
    </div>
  )
}
