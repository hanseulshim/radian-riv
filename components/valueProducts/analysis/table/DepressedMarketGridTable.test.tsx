import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import DepressedMarketGridTable from './DepressedMarketGridTable'

describe('DepressedMarketGridTable', () => {
  let component
  beforeAll(() => {
    const props = {
      tableData: []
    }
    component = shallow(<DepressedMarketGridTable {...props} />)
  })
  it('get table container', () => {
    expect(component.find('div')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
