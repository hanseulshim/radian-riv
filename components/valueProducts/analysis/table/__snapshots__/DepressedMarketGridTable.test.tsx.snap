// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepressedMarketGridTable should match snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": "1em",
    }
  }
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
  />
</div>
`;
